### DISCUSSION TITLE
Configuring API proxy

### DISCUSSION THREAD

**Discussion by jasonwaleryszak** [nuxt/nuxt#17098] [Questions] [answered]
I am attempting to construct an application with a disconnected backend. The frontend (Nuxt) runs on localhost:3000 and the backend (Express.js) runs on localhost:8080.

The current configuration works to some degree. Each of the following requests return the correct results:

GET http://localhost:8080/api/test
GET http://localhost:3000/api/test

The issue, that being infinite loading, occurs when when using `useFetch(...)` or `$fetch(...)` functions. The code below is based on the code provided in https://github.com/nuxt/framework/discussions/4343#discussioncomment-2588462 with only one major change: upgrading http-proxy-middleware from v2 to v3. I tested with v2 and the same issue persisted. 

Not sure if there is a better way to go about this, I will admit I am still rather new to Nuxt 3 (messed around minimally with the pre-releases). Any thoughts would be much appreciated- 

EDIT: The issue appears to relate to `next()` never being called. Not sure how that works under the hood, but that does appear to be part of the issue. 

package.json
```json5
  ...
  "devDependencies": {
    "nuxt": "3.0.0-rc.1"
  },
  "dependencies": {
    "http-proxy-middleware": "^3.0.0-beta.0"
  }
  ...
```

pages/index.vue
```vue
<template>
    ...
</template>

<script setup>
const {data} = await useFetch("/api/test", {method: "GET"});
</script>
```

server/middleware/proxy.js
```js
import {createProxyMiddleware} from 'http-proxy-middleware'

export default async (req, res) => {
    const proxyMiddleware = createProxyMiddleware({
        pathFilter: "/api",
        target: "http://localhost:8080",
        changeOrigin: true
    })

    await new Promise((resolve, reject) => {
        const next = (err) => {
            if (err) {
                reject(err)
            } else {
                resolve()
            }
        }

        proxyMiddleware(req, res, next)
    })
}
```

**Comment by danya-rnd** [comment]
I have the same issue with standalone API. When trying to use useFetch() ot $fetch() I getinfinite loading. As I figured out it happens because next never been called. How to fix this - big question.

Also I have cookie based auth. And cookies are not sent to API throw useFetch.

**Comment by danya-rnd** [comment]
@jasonwaleryszak I know that this one is old. But may be u are still trying to find the solution. My one is - [link](https://github.com/nuxt/framework/discussions/5276)

**Comment by wobsoriano** [comment, marked as answer]
Wrote a simple module for this:

https://github.com/wobsoriano/nuxt-proxy

**Reply by jshimkoski** [reply]
Great work on this. Works like a charm!

**Reply by senher** [reply]
> Wrote a simple module for this:
> 
> https://github.com/wobsoriano/nuxt-proxy

Hi, I introduced this plugin in the project, but it doesn't seem to work properly. If the page uses usefetch get data, the page will keep loading, the data will be fetched normally, the middleware will work normally, but the page cannot be loaded successfully,

**Reply by wobsoriano** [reply]
You have to add the base url for now @senher 

```html
<script setup>
// Base url is required
const { data } = useFetch('http://localhost:3000/api/todos')
</script>
```
